src/wallet/bdk.rs

```diff
@@ -82,7 +82,7 @@ pub struct Wallet {
     pub(crate) store: WalletStore,
 }
 
-/// Speicfy the keychain derivation path from [`HARDENDED_DERIVATION`]
+/// Specify the keychain derivation path from [`HARDENDED_DERIVATION`]
 /// Each kind represents an unhardened index value. Starting with External = 0.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
 pub enum KeychainKind {
```

Comment:
Not relevant in this PR.

Having a doc comment on each of the keychains explaining what they are for would be good. Especially Swapcoins and Contract.